### PR TITLE
Remove SciPy dependency

### PR DESCRIPTION
**Pull request by amylase**
Context: https://github.com/optuna/optuna/pull/4105

Optuna is going to remove its dependency on SciPy. It would be great if we can also remove SciPy dependency from Optuna's dependencies.

PS: I am new to this repository so please let me know if there is something I have to do to get this merged (e.g. benchmarking)

### CODE REVIEW

File: cmaes/_stats.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+            else:
+                b = m
+    else:
+        for _ in range(100):
+            m = (a + b) / 2
+            if math.erfc(math.sqrt(m / 2)) > 1.0 - q:
+                a = m
+            else:
+                b = m
+    return m
```
Comment (c-bata) [inline cmaes/_stats.py:46]: Are these lines taken from Scipy's implementation? If so, we must include scipy's `LICENSE.txt` in this file and our project's readme.

File: setup.cfg [hunk 2]
```diff
@@ -36,9 +36,6 @@ setup_requires =
 install_requires =
     numpy
 
-[options.extras_require]
-cmawm = scipy
-
```
Comment (c-bata) [inline setup.cfg:41]: Perhaps keeping scipy dependency optional and using it if available might be safer as a first step. What do you think?

```python
try:
    from scipy.stats import chi2 as chi2_ppf
    from scipy.stats import norm as norm_cdf
except ImportError:
    from cmaes._stats import chi2_ppf
    from cmaes._stats import norm_cdf
```

**Review by c-bata — COMMENTED** [review]
@amylase Thank you for your pull request! Changes overall looks good.

Let me leave two minor comments about the license and the dependency.

File: cmaes/_stats.py (hunk 1 above)
Comment (amylase) [reply cmaes/_stats.py:46]: these change does not use ScyPy's code.

File: setup.cfg (hunk 2 above)
Comment (amylase) [reply setup.cfg:41]: sounds good. will update.

Comment (c-bata) [reply setup.cfg:41]: Thank you!
By the way, we plan to release the new version at 4pm JST today. If you are busy today, I can work on it in the following pull request later :+1:

Comment (amylase) [reply setup.cfg:41]: sorry I don't think I can do this today.

Comment (c-bata) [reply setup.cfg:41]: No problem! Thank you for letting me know.

I will work on it later 👍

**Review by c-bata — APPROVED** [review]
Thank you again for this great PR.
I will merge this PR with #127.

**Pull request merged**